File: customview/src/main/java/com/example/customview/activity/ProgressBarActivity.java
package com.example.customview.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.example.customview.R;
import com.example.customview.widget.CircleProgressBarView;
import com.example.customview.widget.HorizontalProgressBar;
import com.example.customview.widget.LoadingLineView;
import com.example.customview.widget.LoadingView;
import com.example.customview.widget.ProductProgressBar;
import com.example.customview.widget.StudyPlanProgressView;

import java.util.ArrayList;
import java.util.List;

public class ProgressBarActivity extends AppCompatActivity {
    private static final String TAG = "ProgressBarActivity";
    CircleProgressBarView circleProgressBarView;
    HorizontalProgressBar horizontalProgressBar;
    ProductProgressBar productProgressBar;
    LoadingView loadingView;
    TextView textView;
    LoadingLineView loadingLineView;

    Button button;
    StudyPlanProgressView studyPlanProgressView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pregress_bar);

        circleProgressBarView = findViewById(R.id.circle_progress_view);

        horizontalProgressBar = findViewById(R.id.horizontal_progress_view);
        productProgressBar = findViewById(R.id.product_progress_view);
        loadingView = findViewById(R.id.loading_view);
        loadingLineView = findViewById(R.id.loading_line_view);

        textView = findViewById(R.id.progress_tv);
        button = findViewById(R.id.startAnimationBtn);
        studyPlanProgressView = findViewById(R.id.study_plan_progress_view);

        circleProgressBarView.setProgressWithAnimation(60);
        circleProgressBarView.setProgressListener(new CircleProgressBarView.ProgressListener() {
            @Override
            public void currentProgressListener(float currentProgress) {
                textView.setText("当前进度：" + currentProgress);
            }
        });
        circleProgressBarView.startProgressAnimation();

        horizontalProgressBar.setProgressWithAnimation(60).setProgressListener(new HorizontalProgressBar.ProgressListener() {
            @Override
            public void currentProgressListener(float currentProgress) {
            }
        });
        horizontalProgressBar.startProgressAnimation();

        productProgressBar.setProgress(60).setProgressListener(new ProductProgressBar.ProgressListener() {
            @Override
            public void currentProgressListener(float currentProgress) {
                Log.i(TAG, "currentProgressListener: " + currentProgress);
            }
        });

        loadingView.startAnimation();

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loadingLineView.startLoading();
                loadingView.startAnimation();
                horizontalProgressBar.setProgressWithAnimation(100);
                productProgressBar.setProgress(100);
                circleProgressBarView.setProgressWithAnimation(60).startProgressAnimation();
                circleProgressBarView.setProgressListener(new CircleProgressBarView.ProgressListener() {
                    @Override
                    public void currentProgressListener(float currentProgress) {
                        textView.setText("当前进度：" + currentProgress);
                    }
                });
                studyPlanProgressView.setData(getPlanData(true));
            }
        });
        studyPlanProgressView.setData(getPlanData(false));
    }

    private List<String> getPlanData(Boolean isAll) {
        List<String> list = new ArrayList<>();
        list.add("08月10日");
        list.add("08月11日");
        list.add("08月12日");
        list.add("08月13日");
        if (isAll) {
            list.add("08月14日");
            list.add("08月15日");
            list.add("08月16日");
        }
        return list;
    }

    @Override
    protected void onResume() {
        super.onResume();
        circleProgressBarView.resumeProgressAnimation();
    }

    @Override
    protected void onPause() {
        super.onPause();
        circleProgressBarView.pauseProgressAnimation();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        circleProgressBarView.stopProgressAnimation();
    }
}

File: customview/src/main/java/com/example/customview/activity/ProgressBarActivity.kt
package com.example.customview.activity

import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.customview.databinding.ActivityPregressBarBinding
import com.example.customview.widget.CircleProgressBarView
import com.example.customview.widget.HorizontalProgressBar
import com.example.customview.widget.LoadingLineView
import com.example.customview.widget.LoadingView
import com.example.customview.widget.ProductProgressBar
import com.example.customview.widget.StudyPlanProgressView

/**
 * 用于展示 Progress Bar 功能的 Activity。
 */
open class ProgressBarActivity : AppCompatActivity() {
    private lateinit var binding: ActivityPregressBarBinding
    private lateinit var circleProgressBarView: CircleProgressBarView
    private lateinit var horizontalProgressBar: HorizontalProgressBar
    private lateinit var productProgressBar: ProductProgressBar
    private lateinit var loadingView: LoadingView
    private lateinit var textView: TextView
    private lateinit var loadingLineView: LoadingLineView
    private lateinit var button: Button
    private lateinit var studyPlanProgressView: StudyPlanProgressView

    private val progressListener = CircleProgressBarView.ProgressListener { currentProgress ->
        textView.text = "当前进度：$currentProgress"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityPregressBarBinding.inflate(layoutInflater)
        setContentView(binding.root)

        circleProgressBarView = binding.circleProgressView
        horizontalProgressBar = binding.horizontalProgressView
        productProgressBar = binding.productProgressView
        loadingView = binding.loadingView
        loadingLineView = binding.loadingLineView
        textView = binding.progressTv
        button = binding.startAnimationBtn
        studyPlanProgressView = binding.studyPlanProgressView

        circleProgressBarView.setProgressWithAnimation(60f)
        circleProgressBarView.setProgressListener(progressListener)
        circleProgressBarView.startProgressAnimation()

        horizontalProgressBar
            .setProgressWithAnimation(60f)
            .setProgressListener(HorizontalProgressBar.ProgressListener { })
        horizontalProgressBar.startProgressAnimation()

        productProgressBar
            .setProgress(60f)
            .setProgressListener(ProductProgressBar.ProgressListener { currentProgress ->
                Log.i(TAG, "currentProgressListener: $currentProgress")
            })

        loadingView.startAnimation()

        button.setOnClickListener {
            loadingLineView.startLoading()
            loadingView.startAnimation()
            horizontalProgressBar.setProgressWithAnimation(100f)
            productProgressBar.setProgress(100f)
            circleProgressBarView.setProgressWithAnimation(60f).startProgressAnimation()
            circleProgressBarView.setProgressListener(progressListener)
            studyPlanProgressView.setData(getPlanData(true))
        }
        studyPlanProgressView.setData(getPlanData(false))
    }

    private fun getPlanData(isAll: Boolean): List<String> =
        mutableListOf("08月10日", "08月11日", "08月12日", "08月13日").apply {
            if (isAll) {
                add("08月14日")
                add("08月15日")
                add("08月16日")
            }
        }

    override fun onResume() {
        super.onResume()
        circleProgressBarView.resumeProgressAnimation()
    }

    override fun onPause() {
        super.onPause()
        circleProgressBarView.pauseProgressAnimation()
    }

    override fun onDestroy() {
        super.onDestroy()
        circleProgressBarView.stopProgressAnimation()
    }

    companion object {
        private const val TAG = "ProgressBarActivity"
    }
}
